packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -110,39 +121,47 @@ export class WaferMap extends FoundationElement {
      */
     public render(): void {
         this.renderQueued = false;
-        if (
-            this.canvasSideLength === undefined
-            || this.canvasSideLength === 0
-        ) {
+        this.initalizeInternalModules();
+        this.renderer?.drawWafer();
+    }
+
+    /**
+     * @internal
+     */
+    public queueRender(): void {
+        if (!this.$fastController.isConnected) {
             return;
         }
-        this.renderer?.clearCanvas(
-            this.canvasSideLength,
-            this.canvasSideLength
-        );
-        this.dataManager = new DataManager(
-            this.dies,
-            this.quadrant,
-            { width: this.canvasSideLength, height: this.canvasSideLength },
-            this.colorScale,
-            this.highlightedValues,
-            this.colorScaleMode,
-            this.dieLabelsHidden,
-            this.dieLabelsSuffix,
-            this.maxCharacters
-        );
-        this.renderer = new RenderingModule(this.dataManager, this.canvas);
-        this.zoomHandler = new ZoomHandler(
-            this.canvas,
-            this.zoomContainer,
-            this.dataManager,
-            this.renderer,
-            this.canvasSideLength
-        );
-        this.zoomHandler.attachZoomBehavior();
-        this.renderer.drawWafer();
+        if (!this.renderQueued) {
+            this.renderQueued = true;
+            DOM.queueUpdate(() => this.render());
+        }
+    }
+
+    private initalizeInternalModules(): void {
+        this.eventCoordinator?.detachEvents();
+        this.dataManager = new DataManager(this);
+        this.renderer = new RenderingModule(this);
+        this.eventCoordinator = new EventCoordinator(this);
```

Comment:
Do you know when we will address recreating the event listeners every time there is an update?
We should make sure to track that tech debt if we can't address it in this PR.

Reply:
I would prefer this to be a follow-up PR as there are already other PRs, that we did queue and which depend on this one going in... we can create a github issue for this (if you prefer it this way) or we can track it in our internal AzDO, as part of the this feature. We can set it as tech debt.

Reply:
Please add a link here to where you chose to track it

Reply:
https://dev.azure.com/ni/DevCentral/_workitems/edit/2294259